backend/pkg/contenttype/middleware.go

```diff
@@ -0,0 +1,49 @@
+package contenttype
+
+import (
+	"errors"
+	"mime"
+	"net/http"
+	"slices"
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/mendersoftware/mender-server/pkg/rest.utils"
+)
+
+func CheckJSON() gin.HandlerFunc {
+	return Middleware(false, "application/json")
+}
+
+// For stronger checking with charset, checkCharest should be true.
+// The charset should be uppercase.
+func Middleware(checkCharest bool, contentTypes ...string) gin.HandlerFunc {
+	if len(contentTypes) == 0 {
+		panic("contenttype.Middleware initialized with no parameters")
+	}
+
+	errMsg := "Bad Content-Type, expected "
+	for i, ctype := range contentTypes {
+		if i > 0 {
+			errMsg += ", "
+		}
+		errMsg += "'" + ctype + "'"
+	}
+
+	return func(c *gin.Context) {
+		contentType, params, _ := mime.ParseMediaType(c.GetHeader("Content-Type"))
+		charset, ok := params["charset"]
+
+		if checkCharest && ok {
+			contentType += "; charset=" + charset
+		}
```

Comment:
This assumes that the charset parameter is always the first one. Note that the parameters are position independent so you could theoretically get a header:
```
Content-Type: application/json; foo=bar; charset=UTF-8
```
which has the correct charset with some parameter we don't care about.